src/main/java/com/atlassian/db/replica/api/AuroraConnectionDetails.java

```diff
@@ -10,22 +18,34 @@ private AuroraConnectionDetails(String username, String password) {
         this.password = password;
     }
 
+    public static Builder builder() {
+        return new Builder();
+    }
+
     public String getUsername() {
-        return username;
+        return this.username;
     }
 
     public String getPassword() {
-        return password;
+        return this.password;
+    }
+
+    public ConnectionPoolPerUrlProvider convert() {
+        return url -> () -> DriverManager.getConnection(
```

Comment:
```
DriverManager.getConnection(
            url.toString(),
            username,
            password
```
This could be a default implementation of `ConnectionPoolPerUrlProvider`

Reply:
having this as a default implementation would be misleading since we would not be actually managing any connection pools but creating a new connection every time